pkg/loop/median_service_test.go

```diff
@@ -52,7 +52,7 @@ func TestMedianService_recovery(t *testing.T) {
 	t.Parallel()
 	var limit atomic.Int32
 	median := loop.NewMedianService(logger.Test(t), loop.GRPCOpts{}, func() *exec.Cmd {
-		return helperProcess(loop.PluginMedianName, strconv.Itoa(int(limit.Add(1))))
+		return HelperProcess(loop.PluginMedianName, test.HelperProcessOptions{Limit: int(limit.Add(1))})
```

Comment:
nit/ I am personally a fan of method based constructors, since it is less redundant/verbose for all cases:
```suggestion
		return test.HelperProcessOptions{Limit: int(limit.Add(1))}.New(loop.PluginMedianName)
```
```diff
- return HelperProcess(loop.PluginMedianName, test.HelperProcessOptions{})
+ return NewHelperProcess(loop.PluginMedianName)
```

Reply:
Sounds good :) I've modified this suggestion to take into account the fact that a commandLocation needs to be passed in, but hopefully the API is cleaner now.